### DISCUSSION TITLE
Why do we use wires instead of merging DOFs?

### DISCUSSION THREAD

**Discussion by mrp089** [SimVascular/svZeroDSolver-Archived#67] [Q&A]
In `svZeroDSolver`, we duplicate degrees of freedom (DOFs) when we connect elements and connect them by "wires." This increases the number of DOFs.

Is there a reason we're doing this @JonathanPham @menon-karthik?

**Comment by JonathanPham** [comment]
Where are you seeing this? Which file / which function? 

When I print out the final solution array, I only see unique DoFs.

**Reply by menon-karthik** [reply]
In function "connect_blocks_by_inblock_list" in connections.py, aren't separate wires created on line 89 for connections bA.name_bB.name and bB.name_bA.name, because bA and bB are in each other's connecting_block_list? If so, that would be wires in both directions. 

Then in compute_neq (also in connections.py), number_of_unknowns = 2*len(wire_dict.values()) + block_vars). This doesn't check for duplicate wires, so the DOFs shopuld be getting added up for wires in both directions. But then this also doesn't match the number of equations. The number of equations = num_unknowns only if we aren't duplicating wires somewhere. Not sure where that's happening yet.

**Reply by JonathanPham** [reply]
@menon-karthik and I discussed in person and we learned that two wires are being created between every pair of connected blocks in the `connect_blocks_by_inblock_list` function, but these wires are only in the "forward direction". As such, 2 copies of the same wire end up being created. However, only one of these wires is being stored in the actual `wire_dict` dictionary, since the other one gets overwritten. As such, we have a unique set of wires in the end.
